tests/engine/test_process.py

```diff
@@ -65,7 +65,7 @@ def test_namespaced_process(self):
 
         # Check that the link of the process node has the correct link name
         self.assertTrue('some__name__space__a' in proc.node.get_incoming().all_link_labels())
-        self.assertEqual(proc.node.get_incoming().get_node_by_label('some__name__space__a'), 5)
+        self.assertEqual(proc.node.inputs.some.name.space.a, 5)
```

Comment:
I would keep the original test to make sure that also still works and then simply add the one you added. It would be good to also test `node.inputs['some']['name']['space']['a']

Actually, it would make more sense to add these tests to `tests/orm/utils/test_managers.py`. There are the explicit tests for the `LinkManager` class

Reply:
thanks @sphuber

> I would keep the original test to make sure that also still works and then simply add the one you added. It would be good to also test `node.inputs['some']['name']['space']['a']

added. And should be noticed(and more discuss) that retrieving by `node.inputs.some__name__space__a` not work anymore, which is available previously. This might break the compatibility of some plugins if they use the double underscore to access the node.

> Actually, it would make more sense to add these tests to tests/orm/utils/test_managers.py. There are the explicit tests for the LinkManager class

True, I was trying to add the tests there but it is hard to test in the original testcase. This time I add the test in a new testcase, I hope I did it right and clean.

Reply:
> And should be noticed(and more discuss) that retrieving by node.inputs.some__name__space__a not work anymore, which is available previously. This might break the compatibility of some plugins if they use the double underscore to access the node.

This is a very important point. Is it not possible to keep support for this? Because if so, we can merge this and release it in `v1.6.2` soon. Otherwise this will have to go in v2.0 but even that I am not sure, because we are trying to keep the breaking of plugins in v2.0 minimal. The best option would be to detect if we get a label with double underscores, print a deprecation warning saying to use just normal dots, and then we can remove the support in v2.0

Reply:
> > 
> 
> This is a very important point. Is it not possible to keep support for this? Because if so, we can merge this and release it in `v1.6.2` soon. Otherwise this will have to go in v2.0 but even that I am not sure, because we are trying to keep the breaking of plugins in v2.0 minimal. The best option would be to detect if we get a label with double underscores, print a deprecation warning saying to use just normal dots, and then we can remove the support in v2.0

Thanks a lot @unkcpz and @sphuber for fixing this and the `expose`. This will simplify our work on the plugin side significantly now that the workchain complexity grows. I agree that it would be ideal to keep both. At least until v2.0.